package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -263,14 +268,20 @@ def guess_hydrogens(self, selection='all', max_mass=1.1, min_charge=0.3):
         Alternatively, this function may be used to quickly generate a :class:`str` of potential hydrogen atoms involved
         in hydrogen bonding. This str may then be modified before being used to set the attribute
         :attr:`hydrogens_sel`.
+
+        .. versionchanged: 1.0.0
+            Added `min_mass` parameter to specify minimum mass (Issue #2472)
         """
 
+        assert min_mass < max_mass
+
         ag = self.u.select_atoms(selection)
         hydrogens_ag = ag[
-            np.logical_and(
+            np.logical_and.reduce((
```

Comment:
Maybe being pedantic here, but would it be better to apply to minimum mass filter after the max mass filter?  In terms of number of comparisons you want to apply the most selective filter (max_mass I imagine) first.  Can you check if it's more performant to slice a large atomgroup (to a "medium" size) by max_mass then apply charge, then apply min mass?

Reply:
Do you mean something like the following?
```python
hag = ag[ag.masses < max_mass]
hag = hag[hag.charges > min_charge]
hag = hag[hag.masses > min_mass]
```

On the provided files `waterPSF, waterDCD` this is actually slightly slower (`21.9 µs ± 326 ns` compared to `17.6 µs ± 770`) than the implemented solution, possibly because `np.logical_and.reduce` performs lazy evaluation? In contrast

```python
hydrogens_ag = ag[
            np.logical_and.reduce((
                ag.masses < max_mass,
                ag.charges > min_charge,
                ag.masses > min_mass
            ))
        ]
```
has a comparable speed (`17.7 µs ± 951 ns`). However, `len(ag)==15` which might not be very informative; can you suggest a larger file for this test?

Reply:
Huh you're right, numpy strikes again